src/utils.ts

```diff
@@ -85,10 +105,14 @@ export function execute(
           if (timer) clearTimeout(timer);
         }
       });
-    }).on("error", (err) => {
+    };
+
+    const handleError = (err: Error) => {
       reject(err);
       if (timer) clearTimeout(timer);
-    });
+    };
+
+    const req = https.get(options, handleResponse).on("error", handleError);
```

Comment:
`deno task test` fails, whereas `deno task test:ci` passes, I assume this is because we are using `https` module and it's trying to hit `localhost` with secure protocol (`https`)

```suggestion
    const protocolModule = options.protocol === "https" ? https : http;
    const req = protocolModule.get(options, handleResponse).on("error", handleError);
```